src/yarp.c

```diff
@@ -2059,6 +2059,7 @@ lex_question_mark(yp_parser_t *parser) {
       // TODO: this is wrong. We need to track the state for the lexer on
       // whether or not the ? operator is allowed here. For now, we're depending
       // on space characters.
```

Comment:
I think I still didn't fix this TODO since creating this token still relies on the space, but tokens after this should have a more correct state.